### PR TITLE
(New & Improved) Local Tunnel Debugger API

### PR DESCRIPTION
**Pull request by amsoedal**
Following up on this PR: https://github.com/Azure/vscode-kubernetes-tools-api/pull/31
Implementation: https://github.com/Azure/vscode-kubernetes-tools/pull/851

This PR specifies the new Local Tunnel Debugger API which allows users to redirect traffic from their cluster to their machine when developing, thus combining the benefits of remote and local debugging.  

This change includes the changes & feedback from @daniv-msft's other PR with some name changes and an extra method exposed on the Debugger.

### CODE REVIEW

File: ts/localtunneldebugger/v1.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/**
+ * Supports working with the Debug (Redirect locally) command.
+ */
+export interface LocalTunnelDebuggerV1 {
+    /**
+     * Registers a local tunnel debugger. The debugger will be invoked on user action
+     * when the user selects the Debug (Redirect locally) command in the cluster explorer,
```
Comment (daniv-msft) [inline ts/localtunneldebugger/v1.ts:7]: Nit: please consider renaming these `Redirect locally` in comments as well.

**Review by daniv-msft — APPROVED** [review]
Looks good to me. Approved with a minor comment.

File: ts/localtunneldebugger/v1.ts (hunk 1 above)
Comment (amsoedal) [reply ts/localtunneldebugger/v1.ts]: Thanks for catching that.

**Comment by itowlson** [comment]
@amsoedal Because of other APIs coming in at this release, there were some conflicts that I took the liberty of dealing with rather than going round again.  Hope that was okay.

**Review by itowlson — APPROVED** [review]
This looks great!  Thanks @amsoedal!

**Pull request merged**